presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedScalarToJoin.java

```diff
@@ -58,7 +58,7 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<PlanNode> context)
                 // only scalar subquery wraps expression in EnforceSingleRowNode
                 return new JoinNode(
                         idAllocator.getNextId(),
-                        JoinNode.Type.FULL,
+                        JoinNode.Type.INNER,
```

Comment:
This was a bug actually. And worked incidentally because we converted any 
Join without conditions into NestedLoopJoin (which was not correct for OUTER for case where INNER table was empty).

After changing that in LocalExecutionPlanner:

```
+            if (clauses.isEmpty() && !node.getFilter().isPresent() && node.getType() == INNER) {
                  return createNestedLoopJoin(node, context);
              }
```

this stopped working.